docs/source/transaction_family_specifications/track_and_trade_transaction_family.rst

```diff
@@ -0,0 +1,717 @@
+************************************************
+Track and Trade Transaction Family Specification
+************************************************
+
+Overview
+========
+
+The Track and Trade (T&T) transaction family allows users to track
+goods as they move through a supply chain. Records for goods include a
+history of ownership and custodianship, as well as histories for a
+variety of properties such a temperature and location. These
+properties are managed through a user-specifiable system of record
+types.
+
+
+State
+=====
+
+All T&T objects are serialized using Protocol Buffers before being
+stored in state. These objects include: Agents, Properties
+(accompanied by their auxiliary PropertyPage objects), Proposals,
+Records, and RecordTypes. As described in the Addressing section
+below, these objects are stored in separate sub-namespaces under the
+T&T namespace. To handle hash collisions, all objects are stored in
+lists within protobuf "Container" objects, the order of which is
+described below.
+
+
+Records
+-------
+
+Records represent the goods being tracked by Track and Trade. Almost
+every transaction references some Record.
+
+A Record contains a unique identifier, the name of a RecordType, and
+lists containing the history of its owners and custodians. It also
+contains a `final` flag indicating whether further updates can be made
+to the Record and its Properties. If flag is set to true, then no
+further updates can be made to the Record, including changing its
+`final` flag.
+
+.. code-block:: protobuf
+
+   message Record {
+       message AssociatedAgent {
+           // The Agent's public key
+           string agent_id = 1;
+
+           // Approximately when this agent was associated, as a Unix UTC timestamp
+           uint64 timestamp = 2;
+       }
+
+       // The unique user-defined natural key which identifies the
+       // object in the real world (for example, a serial number)
+       string identifier = 1;
+
+       string record_type = 2;
+
+       // Ordered oldest to newest by timestamp
+       repeated AssociatedAgent owners = 3;
+       repeated AssociatedAgent custodians = 4;
+
+       // Flag indicating whether the Record can be updated. If it is set
+       // to true, then the record has been finalized and no further
+       // changes can be made to it or its Properties.
+       bool final = 5;
+   }
+
+
+Note that while information about a Record's owners and custodians are
+included in the object, information about its Properties are stored
+separately (see below).
```

Comment:
It would be nice to include the section name (as a link to the "Properties" section, I presume).